### PR TITLE
add support for `VK_KHR_multiview`

### LINKED ISSUE #1596: [Feature Request] Multiview/custom RenderPass
I want to make a render pass utilizing VK_KHR_multiview extension, but the current RenderPass implementation doesn't support it.
Since RenderPass is a structure, it's also not possible to create custom one as every function in vulkano expects the concrete type.

Are there any plans to implement multiview support or at least an unsafe constructor to create RenderPass from raw parts?

### PR DESCRIPTION
**Pull request by Timbals**
Supporting the `VK_KHR_multiview` extension enables rendering multiple views in a single renderpass.

Entries for Vulkano changelog:
- Added support for the `VK_KHR_multiview` extension
- Vulkano-shaders: Added support for MultiView SPIR-V capability.
- `multiview` example added showing how to utilize the `VK_KHR_multiview` extension to render to multiple layers of a framebuffer at once

fixes #1596

### CODE REVIEW

**Comment by Rua** [comment]
I noticed you made changes to vk-sys. #1593 changes Vulkano to replace vk-sys with Ash, so you may want to wait for that to merge.

**Comment by Timbals** [comment]
Definitely. I still modified vk-sys because I wrote this two weeks ago but switching to ash should be easy.

File: vulkano/src/render_pass/render_pass.rs [hunk 1]
```diff
@@ -379,10 +379,85 @@ impl RenderPass {
...
+            Some(multiview) => {
+                // TODO don't panic
+
+                assert!(device.loaded_extensions().khr_multiview);
+                assert!(
+                    device
+                        .instance()
+                        .loaded_extensions()
+                        .khr_get_physical_device_properties2
+                );
```
Comment (Rua) [inline vulkano/src/render_pass/render_pass.rs:392]: `khr_multiview` requires this extension to be enabled, to checking for it separately isn't needed. Vulkano doesn't currently check extension dependencies, but I have a PR waiting that adds this (once the Ash migration is merged). So you can write your code to assume that this is checked.

**Comment by Eliah-Lakhin** [comment]
@Timbals fyi, Ash PR is merged. Please let me know when this PR is ready for the final review. Thank you.

File: vulkano/src/render_pass/render_pass.rs (hunk 1 above)
Comment (Rua) [reply vulkano/src/render_pass/render_pass.rs:390]: #1599 will add the dependency check.

Comment (Timbals) [reply vulkano/src/render_pass/render_pass.rs]: I removed the check.

**Comment by Timbals** [comment]
@Eliah-Lakhin ready for review

**Comment by Eliah-Lakhin** [comment]
@Timbals Looks good to me. An example produces correct result. I'm proceeding with merging...

**Comment by Eliah-Lakhin** [comment]
PR merged. Thank you for your work!

**Pull request merged**